src/main/java/com/auth0/jwk/UrlJwkProvider.java

```diff
@@ -125,9 +128,14 @@ static URL urlForDomain(String domain) {
     @Override
     public Jwk get(String keyId) throws JwkException {
         final List<Jwk> jwks = getAll();
-        for (Jwk jwk: jwks) {
-            if (keyId.equals(jwk.getId())) {
-                return jwk;
+        if (keyId == null && jwks.size() == 1) {
+            return jwks.get(0);
+        }
+        if (keyId != null) {
+            for (Jwk jwk : jwks) {
+                if (keyId.equals(jwk.getId())) {
```

Comment:
Seems odd that this portion of the method takes into account multiple entries in the JWKs but the previous one does not.

Reply:
it's only filtering the entries in a list for those which match a given key id. see:

- if key is specified, match by key even if set size is 1, since the user wants that key.
- if key is not specified and size is 1 the solution is trivial -> return that item
- in any other case, throw key with id "{id}" not found.